browser/components/EditProject.tsx

```diff
@@ -0,0 +1,315 @@
+      numUsers: number;
+      projects: number[];
+      sponsor: string;
+    }
+  ];
+  project: string;
+}
+
+interface State {
+  alert: any;
+  groupList: [
+    {
+      goal: string;
+      group_id: number;
+      group_name: string;
+      contribMTD: number;
+      contribMonth: number;
+      contribWeek: number;
+      contribYear: number;
+      numGroups: number;
+      numUsers: number;
+      projects: number[];
+      sponsor: string;
+    }
+  ];
+  projectGroups: string;
+  projectID: number;
+  projectInfo?: {
+    groups: [
+      {
+        contribMTD: number;
+        contribMonth: number;
+        contribWeek: number;
+        contribYear: number;
+        goal: string;
+        group_id: number;
+        group_name: string;
+        projects: number[];
+        sponsor: string;
+      }
+    ];
+    project: {
+      project_auto_aprovable: boolean;
+      project_id: number;
+      project_license: string;
+      project_name: string;
+      project_url: string;
+      project_verified: boolean;
+    };
+    users: [
+      {
+        company_alias: string;
+        contribMTD: number;
+        contribMonth: number;
+        contribWeek: number;
+        contribYear: number;
+        github_alias: string;
+      }
+    ];
+  };
+  projectLicense: string;
+  projectName: string;
+  projectURL: string;
+  projectUsers: string;
+}
+
+export default class EditProject extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      alert: null,
+      projectGroups: '',
+      projectID: null,
+      projectInfo: null,
+      projectLicense: '',
+      projectName: '',
+      projectURL: '',
+      projectUsers: '',
+      groupList: null,
+    };
+  }
+
+  async componentDidUpdate(prevProps) {
+    if (this.props !== prevProps) {
+      await this.getProject();
+
+      const projectGroups = [];
+      for (const group of this.state.projectInfo.groups) {
+        projectGroups.push(group.group_id);
+      }
+
+      const projectUsers = [];
+      let projectDetails = {
+        project_id: null,
+        project_license: null,
+        project_name: null,
+        project_url: null,
+      };
+      if (this.state.projectInfo) {
+        projectDetails = this.state.projectInfo.project;
+        for (const user of this.state.projectInfo.users) {
+          projectUsers.push(user.company_alias);
+        }
+      }
+
+      this.setState({
+        projectGroups: projectGroups.join(','),
+        projectID: projectDetails.project_id || '',
+        projectLicense: projectDetails.project_license || '',
+        projectName: projectDetails.project_name || '',
+        projectURL: projectDetails.project_url || '',
+        projectUsers: projectUsers.join(','),
+      });
+    }
+  }
+
+  getProject = async () => {
+    if (this.props.project) {
+      const projectInfo = await reqJSON(
+        `/api/strategic/projects/${this.props.project}`
+      );
+      this.setState({
```

Comment:
Watch out; this is asynchronous. It's entirely possible getProject will return/resolve before the state change occurred. Relying on state afterwards in the previous function might not yield the results you want.

https://reactjs.org/docs/react-component.html#setstate